libs/testserver/fake_workspace.go

```diff
@@ -247,6 +252,52 @@ func (s *FakeWorkspace) JobsGet(jobId string) Response {
 	}
 }
 
+func (s *FakeWorkspace) JobsRunNow(jobId int64) Response {
+	defer s.LockUnlock()()
+
+	_, ok := s.jobs[jobId]
+	if !ok {
+		return Response{
+			StatusCode: 404,
+		}
+	}
+
+	runId := s.nextJobRunId
+	s.nextJobRunId++
+	s.jobRuns[runId] = jobs.Run{
+		RunId: runId,
+		State: &jobs.RunState{
+			LifeCycleState: jobs.RunLifeCycleStateRunning,
+		},
+		RunPageUrl: fmt.Sprintf("%s/job/run/%d", s.url, runId),
+		RunType:    jobs.RunTypeJobRun,
+		RunName:    "run-name",
+	}
+
+	return Response{
+		Body: jobs.RunNowResponse{
+			RunId: runId,
+		},
+	}
+}
+
+func (s *FakeWorkspace) JobsGetRun(runId int64) Response {
+	defer s.LockUnlock()()
+
+	run, ok := s.jobRuns[runId]
+	if !ok {
+		return Response{
+			StatusCode: 404,
+		}
+	}
+
+	// Mark the run as terminated.
+	run.State.LifeCycleState = jobs.RunLifeCycleStateTerminated
```

Comment:
This seems to be very specific to a particular test, might not be reusable. Perhaps instead you can have explicit endpoint to update this field?

Reply:
OK, we can extend this later.